Add initial list of marker genes for Ewing sarcoma tumor cells

allyhawkins opened the pull request:
### Purpose/implementation Section

#### Please link to the GitHub issue that this pull request addresses.

Closes #373 

#### What is the goal of this pull request?

This PR adds in a TSV file with a list of marker genes that have been used to classify Ewing sarcoma tumor cells as tumor cells in the literature. The goal is to be able to use this list of marker genes to assist in identification of tumor cells in the Ewing's samples. 

#### Briefly describe the general approach you took to achieve this goal.

To create this TSV file I looked for marker genes that were used to identify tumor cells in published single-cell RNA-seq studies of Ewing sarcoma tumors. I went through the list of publications that I had compiled in https://github.com/AlexsLemonade/OpenScPCA-analysis/discussions/292, which all had single-cell RNA-seq of either Ewing's cell lines or patient tumors. I ended up focusing on the two studies that do single-cell RNA-seq on patient tumors rather than cell lines. Those papers had distinct lists of genes that were used to identify tumor cells, many of those genes fit with my expectations of what the markers would be based on my biological knowledge about the disease. The other papers I think will be more helpful when creating marker gene lists for cell states within the tumor cells.  

For https://doi.org/10.1158/2767-9764.CRC-23-0027, I used the genes listed in [Supplemental Methods under `Cell annotation using external data`](https://aacr.silverchair-cdn.com/aacr/content_public/journal/cancerrescommun/3/10/10.1158_2767-9764.crc-23-0027/1/crc-23-0027-s01.pdf?Expires=1716815671&Signature=JgkH5QWs1J0dVeZFTCnZzfi2eDDQHhGsYazaF2UGI5vhPIatv7egTlW-3974FeZ~hM5Ac0gqiq0oBqeenDNWgN0PbU30N9-8n0LrsUmLHA2kCzhBxiHv8D-eKPXL0dSA9H3NKb~PkqTvOeV2QSde~MpCQL~cFh2Yn2Uf9XtAwrXrGhL3nOLIdb~qWBWtQUf1u0w~zynaDsz1hMh32gV5dEQVvZ0XZ4ozc9NO8neUbNvRJk5SSgAIPmQk1WXHKZpIheUbGU2VI2HeggkprVHO0lUELcgb1MmtziupgAkjE-xRrRUtbPVLe8QTWny6QtgJ6IcoH120LbFtRIfL9nOpuA__&Key-Pair-Id=APKAIE5G5CRDK6RD3PGA). They mention other genes in the text and figures that were highly expressed in tumor cells, but I thought it would be better to start with the specific list they mention in the methods. 

For https://doi.org/10.1101/2024.01.18.576251, they mention a core set of genes in the text and then point to Supplemental Figure 2C that contains the genes that were unique to Ewing sarcoma cells. They mention different genes that are associated with different cell states throughout the text, but I chose to only use the ones that were expressed across all Ewing's cells over other normal cells in the samples. 

One thing to keep in mind is that both papers used multiple methods to validate tumor cell identification, which included marker genes, CNV inference, and expression of EWS/FLI1 target genes. There are a few EWS signatures in MsigDB that we might want to use to help validate tumor cells. However, there's also a lot of literature that notes differing expression of EWS targets in tumor cells, including some tumor cells that have a repressed EWS signature, so we should proceed with caution there.

#### If known, do you anticipate filing additional pull requests to complete this analysis module?

The next PR should be to start the exploration notebook and address #374. 


### Author checklists

#### Analysis module and review

- [x] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [ ] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [ ] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

jashapiro commented on analyses/ewings-cell-type/references/tumor_marker_genes.tsv:
```diff
@@ -0,0 +1,12 @@
+cell_type	gene_symbol	ensembl_gene_id	source
+tumor	CD99	ENSG00000002586	10.1158/2767-9764.CRC-23-0027
```
In looking at the references, `CD99` was also found in 10.1101/2024.01.18.576251? I don't know if you want to have multiple sources on each line, or repeated lines, but I think you should probably capture when a gene is supported by more than one reference somehow. (I'd probably do the tidier repeated lines with 1 ref per line, and then deduplicate in code)

Update: I see now where you got this specific list, which does not include CD99, though there is a slightly later sentence:
> We found that compared to other cell types in the primary tumor, Ewing cells across all samples express moderate to high levels of PAPPA but low levels of most HLA genes. For immunotherapeutic targets, Ewing sarcoma cells uniquely express higher levels of ADGRG2, IL1RAP, and LINGO1. CD99 and STEAP1 is also highly expressed in Ewing sarcoma cells, although there is significant expression in CAFs as well.

I don't know if you want to include any of those extra genes too? There is also a longer list in Supp table 2, but I don't understand how that list was reduced to the lists in text

jashapiro replied:
From 10.1101/2024.01.18.576251: "Among the Ewing sarcoma cells, a proliferating population was identified by the expression of MKI67 and PCNA (Fig. 1C-D)." Would you want to include such subtyping?

allyhawkins replied:
> Update: I see now where you got this specific list, which does not include CD99, though there is a slightly later sentence:

So CD99 has been the historic marker of Ewing's cells, but it is also expressed at a slightly lower level in normal cells. Fig 1F in 10.1101/2024.01.18.576251 shows that in this particular set of patients its expressed at varying levels across all cells. I got the list I used for this paper specifically from Fig 1C because there are a lot of sentences in the text that mention various genes that were used. However, looking at 1F a lot of those genes are either only in a subset of patients. 

> I don't know if you want to include any of those extra genes too? There is also a longer list in Supp table 2, but I don't understand how that list was reduced to the lists in text

I'm trying to keep this initial list relatively short and only stick to genes that are consistently expressed across multiple patients. I know this is hard because the dataset we have is probably going to have differences, but I want to use CellAssign and the longer the list the more resources that takes. Looking at Figure 1F, maybe we add in ADGRG2, but LINGO1 and IL1RAP are not consistently expressed. 

> From 10.1101/2024.01.18.576251: "Among the Ewing sarcoma cells, a proliferating population was identified by the expression of MKI67 and PCNA (Fig. 1C-D)." Would you want to include such subtyping?

Yes definitely! In this initial list I just wanted to do things that marked tumor cells, but then we should create a list that marks tumor cell states. There are a lot more cell states that have been identified besides this, so that will be a more comprehensive list and I think should be part of a separate review.

jashapiro replied:
This all makes sense! I wrote all that as I was reviewing, and I agree that a lot of it is for potential future directions. I agree especially with keeping the initial list compact. 

As I said in the overall comments, the main thing is to have some documentation of the decisions in the repo itself!

jashapiro reviewed (commented):
This looks like a good start. I think the main thing I would suggest is that you add some of the information that you put into the PR text about how you decided which sets of genes to include somewhere in the repository itself. That might be a methods or readme document in the `references` directory or something like that? Somebody coming back to this (you, writing it up) is going to have a lot of questions about how you parsed each reference, and having to search github to find your logic is not going to be very user-friendly!

allyhawkins commented:
Thanks for the review @jashapiro. 

- I changed to using the full URLs for the source of the marker genes. 
- I did end up adding another line for CD99 since it is in both papers. 
- I also added in ADGRG2 based on Fig. 1F. 
- And then I added a README file for the references directory, which includes a description of these marker genes. 

This should be ready for another look!

jashapiro commented on analyses/ewings-cell-type/references/README.md:
```diff
@@ -0,0 +1,16 @@
+
+The marker gene list was created using gene lists from two publications:
+
+1. https://doi.org/10.1158/2767-9764.CRC-23-0027 - Marker genes were obtained from the list in [Supplemental Methods under `Cell annotation using external data`](https://aacr.silverchair-cdn.com/aacr/content_public/journal/cancerrescommun/3/10/10.1158_2767-9764.crc-23-0027/1/crc-23-0027-s01.pdf?Expires=1716815671&Signature=JgkH5QWs1J0dVeZFTCnZzfi2eDDQHhGsYazaF2UGI5vhPIatv7egTlW-3974FeZ~hM5Ac0gqiq ...
```
Maybe put a short reference here as well? I also checked, you don't need the full url you pasted in. Most of that seems to be for tracking/login. A good tip for urls like this is to see if you can delete everything from the `?` onward. Usually you can.
```suggestion
1. Visser _et al._ (2023) https://doi.org/10.1158/2767-9764.CRC-23-0027 - Marker genes were obtained from the list in [Supplemental Methods under `Cell annotation using external data`](https://aacr.silverchair-cdn.com/aacr/content_public/journal/cancerrescommun/3/10/10.1158_2767-9764.crc-23-0027/1/crc-23-0027-s01.pdf).
```

jashapiro commented on analyses/ewings-cell-type/references/README.md:
```diff
@@ -0,0 +1,16 @@
+
+1. https://doi.org/10.1158/2767-9764.CRC-23-0027 - Marker genes were obtained from the list in [Supplemental Methods under `Cell annotation using external data`](https://aacr.silverchair-cdn.com/aacr/content_public/journal/cancerrescommun/3/10/10.1158_2767-9764.crc-23-0027/1/crc-23-0027-s01.pdf?Expires=1716815671&Signature=JgkH5QWs1J0dVeZFTCnZzfi2eDDQHhGsYazaF2UGI5vhPIatv7egTlW-3974FeZ~hM5Ac0gqiq ...
+
+2. https://doi.org/10.1101/2024.01.18.576251 - Marker genes were determined to be those that were unique to Ewing sarcoma cells in Figure 1C and 1F.
```
```suggestion
2. Goodspeed _et al._ (2024) https://doi.org/10.1101/2024.01.18.576251 - Marker genes were determined to be those that were unique to Ewing sarcoma cells in Figure 1C and 1F.
```

jashapiro reviewed (approved):
LGTM, with just a little suggestion to add an easier reference for readers and one to remove tokens from a url.
